### DISCUSSION TITLE
Best way to configure unique usernames?

### DISCUSSION THREAD

**Discussion by rminami** [supabase/supabase#1976] [Questions] [answered]
Hi, I'm trying to figure out the best way to configure unique usernames in Supabase. 

So far, I've been putting all of my user data into `user_metadata` through [`auth.update()`](https://supabase.io/docs/reference/javascript/auth-update), but for usernames, I want to make sure they are unique across the entire project, and this method doesn't seem to offer a way to do this.

Do I have to create a separate table and store user ids / usernames there? This seems like a common use case, so I'm wondering if there was anything simpler I could do.

Thanks!

**Comment by silentworks** [comment, marked as answer]
It's probably best to store any user data inside of a separate table. I have a table called users where I store such information. You can then create set the username column to be unique when creating the table.

You can use a SQL query like this inside of the query viewer in the Supabase dashboard to handle this:

```sql
ALTER TABLE ONLY "users"
  ADD CONSTRAINT unique_username UNIQUE ("username");
```

This will throw an error when trying to create a user with the same username that already exists in your database, you can catch this in your code and give the user the necessary feedback then.

**Reply by rminami** [reply]
I suppose I'll have to do that then. Thanks for the help!
